stan/math/prim/mat/fun/to_matrix.hpp

```diff
@@ -111,7 +100,162 @@ namespace stan {
+     * order with the specified number of rows and columns.
+     *
+     * @tparam T type of the scalar
+     * @param x vector of values
+     * @param m rows
+     * @param n columns
+     * @return the matrix representation of the input
+     * @throw <code>std::invalid_argument</code>
+     * if the sizes do not match
+     */
+    template <typename T>
+    inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const std::vector<T>& x, int m, int n) {
+      static const char* fun = "to_matrix(array)";
+      check_size_match(fun, "rows * columns", m * n, "vector size",
+                       x.size());
+      return Eigen::Map<const
+                        Eigen::Matrix<T, Eigen::Dynamic,
+                                      Eigen::Dynamic> >(&x[0], m, n);
+    }
+
+    /**
+     * Returns a matrix representation of the vector in column-major
+     * order with the specified number of rows and columns.
+     *
+     * @param x vector of values
+     * @param m rows
+     * @param n columns
+     * @return the matrix representation of the input
+     * @throw <code>std::invalid_argument</code>
+     * if the sizes do not match
+     */
+    inline Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const std::vector<int>& x, int m, int n) {
+      static const char* fun = "to_matrix(array)";
+      int size = x.size();
+      check_size_match(fun, "rows * columns", m * n,
+                       "vector size", size);
+      Eigen::Matrix<double,
+                    Eigen::Dynamic, Eigen::Dynamic> result(m, n);
+      double* datap = result.data();
+      for (int i=0; i < size; i++)
+        datap[i] = x[i];
+      return result;
+    }
+
+    /**
+     * Returns a matrix representation of the vector in column-major
+     * order with the specified number of rows and columns.
+     *
+     * @tparam T type of the scalar
+     * @param x matrix
+     * @param m rows
+     * @param n columns
+     * @param cm column-major indicator:
+     * if 1, output matrix is transversed in column-major order,
+     * if 0, output matrix is transversed in row-major order,
+     * otherwise function throws std::invalid_argument
+     * @return Reshaped inputted matrix
+     * @throw <code>std::invalid_argument</code>
+     * if the sizes do not match
+     */
+    template <typename T, int R, int C>
+    inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const Eigen::Matrix<T, R, C>& x, int m, int n, int cm) {
+      if (cm == 1) {
+        return to_matrix(x, m, n);
+      } else if (cm == 0) {
+        check_size_match("to_matrix", "rows * columns", m * n,
+                         "matrix size", x.size());
+        Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+          result(m, n);
+        for (int i=0, ij=0; i < m; i++)
+          for (int j=0; j < n; j++, ij++)
+            result(i, j) = x(ij);
+        return result;
+      } else {
```

Comment:
Then you can get rid of this test because the `bool` will have to be 0 or 1.  And you can replace `if (cm == 1)` with just `if (cm)`, though I'd prefer `col_major` as the variable name so it's easier to understand (no big deal if you like `cm` better).